src/extension/api/search.ts

```diff
@@ -0,0 +1,26 @@
+import { Unsubscribable } from 'rxjs'
+import { QueryTransformerProvider } from 'sourcegraph'
+import { SearchAPI } from 'src/client/api/search'
+import { ProviderMap } from './common'
+
+/** @internal */
+export interface ExtSearchAPI {
+    $transformQuery: (id: number, query: string) => Promise<string | null | undefined>
```

Comment:
As mentioned above, I think you should only accept `Promise<string>`, not `Promise<string | null | undefined>`. I'm not sure what null and undefined mean without checking docs or impl, and it's not clear that there is any benefit to allowing those return types.